python/ray/tests/test_object_spilling.py

```diff
@@ -605,5 +609,53 @@ def get_object(arr):
     assert hash_value == hash_value1
 
 
+@pytest.mark.skipif(
+    platform.system() == "Windows", reason="Failing on Windows.")
+def test_spill_dir_cleanup_on_raylet_start(object_spilling_config):
+    object_spilling_config, temp_folder = object_spilling_config
+    cluster = Cluster()
+    cluster.add_node(
+        num_cpus=0,
+        object_store_memory=75 * 1024 * 1024,
+        _system_config={"object_spilling_config": object_spilling_config})
+    ray.init(address=cluster.address)
+    node2 = cluster.add_node(num_cpus=1, object_store_memory=75 * 1024 * 1024)
+
+    # This task will run on node 2 because node 1 has no CPU resource
+    @ray.remote(num_cpus=1)
+    def run_workload():
+        ids = []
+        for _ in range(2):
+            arr = np.random.rand(5 * 1024 * 1024)  # 40 MB
+            ids.append(ray.put(arr))
+        return ids
+
+    ids = ray.get(run_workload.remote())
+    assert not is_dir_empty(temp_folder)
+
+    # Kill node 2
+    raylet_process = node2.all_processes[ray_constants.PROCESS_TYPE_RAYLET][
+        0].process
+    raylet_process.terminate()
+    raylet_process.wait()
+
+    # Verify that the spill folder is not empty
+    assert not is_dir_empty(temp_folder)
+
+    import time
+    time.sleep(2)
+    # Start a new node
+    cluster.add_node(num_cpus=1, object_store_memory=75 * 1024 * 1024)
+    time.sleep(2)
```

Comment:
Why do we need sleep here? Process should've been killed at this point as we have raylet_process.wait() up there right?